pkg/controller/instance/instance_controller.go

```diff
@@ -19,94 +19,55 @@ import (
-	"sigs.k8s.io/controller-runtime/pkg/manager"
-	"sigs.k8s.io/controller-runtime/pkg/predicate"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
-	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
-// Add creates a new Instance Controller and adds it to the Manager with default RBAC.
-//
-// The Manager will set fields on the Controller and start it when the Manager is started.
-func Add(mgr manager.Manager) error {
-	log.Printf("InstanceController: Registering instance controller.")
-	return add(mgr, newReconciler(mgr))
-}
-
-// newReconciler returns a new reconcile.Reconciler
-func newReconciler(mgr manager.Manager) reconcile.Reconciler {
-	return &ReconcileInstance{Client: mgr.GetClient(), scheme: mgr.GetScheme(), recorder: mgr.GetEventRecorderFor("instance-controller")}
+// Reconciler reconciles an Instance object.
+type Reconciler struct {
+	client.Client
+	Recorder record.EventRecorder
+	Scheme   *runtime.Scheme
 }
 
-// add adds a new Controller to mgr with r as the reconcile.Reconciler.
-func add(mgr manager.Manager, r reconcile.Reconciler) error {
-	// Create a new controller
-	c, err := controller.New("instance-controller", mgr, controller.Options{Reconciler: r})
-	if err != nil {
-		return err
-	}
-
-	// Watch for changes to Instance
-	if err = c.Watch(&source.Kind{Type: &kudov1alpha1.Instance{}}, &handler.EnqueueRequestForObject{}, instanceEventFilter(mgr)); err != nil {
-		return err
-	}
-
-	// Watch for changes to OperatorVersion. Since changes to OperatorVersion and Instance are often happening
-	// concurrently there is an inherent race between both update events so that we might see a new Instance first
-	// without the corresponding OperatorVersion. We additionally watch OperatorVersions and trigger
-	// reconciliation for the corresponding instances.
-	//
-	// Define a mapping from the object in the event (OperatorVersion) to one or more objects to
-	// reconcile (Instances). Specifically this calls for a reconciliation of any owned objects.
-	ovEventHandler := handler.ToRequestsFunc(
+// SetupWithManager registers this reconciler with the controller manager
+func (r *Reconciler) SetupWithManager(
+	mgr ctrl.Manager) error {
+	addOvRelatedInstancesToReconcile := handler.ToRequestsFunc(
 		func(a handler.MapObject) []reconcile.Request {
 			requests := make([]reconcile.Request, 0)
-			// We want to query and queue up operators Instances
 			instances := &kudov1alpha1.InstanceList{}
 			// we are listing all instances here, which could come with some performance penalty
 			// a possible optimization is to introduce filtering based on operatorversion (or operator)
 			err := mgr.GetClient().List(
 				context.TODO(),
 				instances,
 			)
-
 			if err != nil {
 				log.Printf("InstanceController: Error fetching instances list for operator %v: %v", a.Meta.GetName(), err)
 				return nil
 			}
-
 			for _, instance := range instances.Items {
 				// Sanity check - lets make sure that this instance references the operatorVersion
```

Comment:
there is no details on:
1. why we need to do a sanity check
2. or what happens if things are not sane..  how and where is that handled?   I assume that is an upgrade. (under normal conditions)

Reply:
I actually think the comment does not make sense, that is no sanity check, since we filter all instances, we need to then pick only those that match the OV we're reconciling... I've changed the comment

Reply:
It's old code, but you're right: the word "sanity check"  is misleading - it's a "normal check" since we list _all_ instances.